src/client/mod.rs

```diff
@@ -941,45 +943,59 @@ impl Client {
 /// // ensure a valid token is in fact valid:
 /// assert!(validate_token("Mjg4NzYwMjQxMzYzODc3ODg4.C_ikow.j3VupLBuE1QWZng3TMGH0z_UAwg").is_ok());
 ///
-/// // "cat" isn't a valid token:
-/// assert!(validate_token("cat").is_err());
-///
-/// // tokens must have three parts, separated by periods (this is still
-/// // actually an invalid token):
-/// assert!(validate_token("aaa.abcdefgh.bbb").is_ok());
-///
-/// // the second part must be _at least_ 6 characters long:
-/// assert!(validate_token("a.abcdef.b").is_ok());
-/// assert!(validate_token("a.abcde.b").is_err());
+/// // helpful to prevent typos
+/// assert!(validate_token("Njg4NzYwMjQxMzYzODc3ODg4.C_ikow.j3VupLBuE1QWZng3TMGH0z_UAwg").is_err());
 /// ```
 ///
 /// # Errors
 ///
 /// Returns a [`ClientError::InvalidToken`] when one of the above checks fail.
 /// The type of failure is not specified.
 pub fn validate_token(token: impl AsRef<str>) -> Result<()> {
-    let token = token.as_ref();
+    let (user_id, generation_time) =
+        parse_token(token.as_ref()).ok_or(Error::Client(ClientError::InvalidToken))?;
 
-    if token.is_empty() {
+    dbg!(user_id, generation_time);
+
+    // Check if timestamps are in a sensible range
+    if user_id.created_at().year() >= 2100 || generation_time.year() >= 2100 {
         return Err(Error::Client(ClientError::InvalidToken));
     }
 
+    Ok(())
+}
+
+/// Verifies that the token adheres to the Discord token format and extracts the bot user ID and the
+/// token generation timestamp
+fn parse_token(token: &str) -> Option<(UserId, chrono::NaiveDateTime)> {
```

Comment:
```suggestion
pub fn parse_token(token: impl AsRef<str>) -> Option<(UserId, chrono::NaiveDateTime)> {
```

Reply:
So you're suggesting to make that method public. In that case would it be good to make a designated struct?
```rust
pub struct TokenComponents {
    bot_user_id: UserId,
    token_creation_time: chrono::NaiveDateTime,
}
```
That way, callers of the function can make it explicit what the "UserId" and the "NaiveDateTime" actually represent.